tests/unit/test_diffsync_model_flags.py

```diff
@@ -0,0 +1,236 @@
+    backend_a_with_extra_models.get(
+        backend_a_with_extra_models.device, "nyc-spine3"
+    ).model_flags |= DiffSyncModelFlags.SKIP_UNMATCHED_SRC
+    diff = backend_a.diff_from(backend_a_with_extra_models)
+    assert diff.summary() == {"create": 0, "update": 0, "delete": 0, "no-change": 23}
+
+
+def test_diffsync_sync_with_skip_unmatched_src_flag_on_models(backend_a, backend_a_with_extra_models):
+    backend_a_with_extra_models.get(
+        backend_a_with_extra_models.site, "lax"
+    ).model_flags |= DiffSyncModelFlags.SKIP_UNMATCHED_SRC
+    backend_a_with_extra_models.get(
+        backend_a_with_extra_models.device, "nyc-spine3"
+    ).model_flags |= DiffSyncModelFlags.SKIP_UNMATCHED_SRC
+
+    backend_a.sync_from(backend_a_with_extra_models)
+
+    # New objects should not have been created
+    with pytest.raises(ObjectNotFound):
+        backend_a.get(backend_a.site, "lax")
+    with pytest.raises(ObjectNotFound):
+        backend_a.get(backend_a.device, "nyc-spine3")
+    assert "nyc-spine3" not in backend_a.get(backend_a.site, "nyc").devices
+
+    diff = backend_a.diff_from(backend_a_with_extra_models)
+    assert diff.summary() == {"create": 0, "update": 0, "delete": 0, "no-change": 23}
+
+
+def test_diffsync_diff_with_skip_unmatched_dst_flag_on_models(backend_a, backend_a_minus_some_models):
+    # Validate that there are 3 extras objects out of the box
+    diff = backend_a.diff_from(backend_a_minus_some_models)
+    assert diff.summary() == {"create": 0, "update": 0, "delete": 12, "no-change": 11}
+
+    # Check that only the device "rdu-spine1" and its 2 interfaces are affected by the flag
+    backend_a.get(backend_a.device, "rdu-spine1").model_flags |= DiffSyncModelFlags.SKIP_UNMATCHED_DST
+    diff = backend_a.diff_from(backend_a_minus_some_models)
+    assert diff.summary() == {"create": 0, "update": 0, "delete": 9, "no-change": 11}
+
+    # Check that only one additional device "sfo-spine2" and its 3 interfaces are affected by the flag
+    backend_a.get(backend_a.device, "sfo-spine2").model_flags |= DiffSyncModelFlags.SKIP_UNMATCHED_DST
+    diff = backend_a.diff_from(backend_a_minus_some_models)
+    assert diff.summary() == {"create": 0, "update": 0, "delete": 5, "no-change": 11}
+
+
+def test_diffsync_sync_with_skip_unmatched_dst_flag_on_models(backend_a, backend_a_minus_some_models):
+    backend_a.get(backend_a.site, "rdu").model_flags |= DiffSyncModelFlags.SKIP_UNMATCHED_DST
+    backend_a.get(backend_a.device, "sfo-spine2").model_flags |= DiffSyncModelFlags.SKIP_UNMATCHED_DST
+    backend_a.sync_from(backend_a_minus_some_models)
+
+    # Objects should not have been deleted
+    # rdu-spine1 hasn't been deleted because its parent hasn't been deleted
+    assert backend_a.get(backend_a.site, "rdu") is not None
+    assert backend_a.get(backend_a.device, "rdu-spine1") is not None
+    assert backend_a.get(backend_a.device, "sfo-spine2") is not None
```

Comment:
Also verify that sfo-spine2's child interfaces aren't deleted perhaps?

Reply:
Done